src/manifest/__tests__/period.test.ts

```diff
@@ -190,28 +176,58 @@ describe("Manifest - Period", () => {
     });
     expect(period.parsingErrors).toHaveLength(1);
 
-    expect(adaptationSpy).toHaveBeenCalledTimes(1);
-    expect(adaptationSpy).toHaveBeenCalledWith(videoAda1, undefined);
+    expect(adaptationSpy).toHaveBeenCalledTimes(2);
+    expect(adaptationSpy).toHaveReturnedTimes(1);
+    expect(adaptationSpy).toHaveBeenCalledWith(videoAda1, {});
+    expect(adaptationSpy).toHaveBeenCalledWith(barAda1, {});
+    expect(adaptationSpy).toHaveReturnedWith(period.adaptations.video[0]);
 
     const [error] = period.parsingErrors;
     expect(error).toBeInstanceOf(Error);
     expect(error.code).toBe("MANIFEST_UNSUPPORTED_ADAPTATION_TYPE");
     expect(error.type).toBe("MEDIA_ERROR");
     expect(error.fatal).toEqual(false);
-    expect(logSpy).toHaveBeenCalledTimes(1);
+  });
+
+  /* tslint:disable:max-line-length */
+  it("should throw if the adaptation throws for another reason than an unsupported type", () => {
+  /* tslint:enable:max-line-length */
+    const adaptationSpy = jest.fn(arg => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 158. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=432&issues=AWgy3w-nSi5GWQS98kk0&open=AWgy3w-nSi5GWQS98kk0)

[ISSUE_KEY]: AWgy3w-nSi5GWQS98kk0